classes/sota_sanity.bbclass

```diff
@@ -10,6 +10,48 @@ def sota_check_required_variables(status, d):
         if not d.getVar(var):
             status.addresult("%s should be set in your local.conf.\n" % var)
 
+def sota_check_variables_validity(status, d):
+    import re
+    import os.path
+
+    if d.getVar("OSTREE_BRANCHNAME") and re.match("^[a-zA-Z0-9_-]*$", d.getVar("OSTREE_BRANCHNAME")) is None:
+        status.addresult("OSTREE_BRANCHNAME Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if d.getVar("SOTA_HARDWARE_ID") and re.match("^[a-zA-Z0-9_-]*$", d.getVar("SOTA_HARDWARE_ID")) is None:
+        status.addresult("SOTA_HARDWARE_ID Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if d.getVar("SOTA_CLIENT_FEATURES") is not None:
+        for feat in d.getVar("SOTA_CLIENT_FEATURES").split(' '):
+            if feat not in ("hsm", "serialcan", "ubootenv", ""):
+                status.addresult("SOTA_CLIENT_FEATURES should only include hsm, serialcan and bootenv.\n")
+                break
+    if d.getVar("SOTA_CLIENT_PROV") is not None:
+        prov = d.getVar("SOTA_CLIENT_PROV").strip()
+        if prov not in ("aktualizr-shared-prov", "aktualizr-device-prov", "aktualizr-device-prov-hsm", ""):
+            status.addresult("Valid options for SOTA_CLIENT_PROV are aktualizr-shared-prov, aktualizr-device-prov and aktualizr-device-prov-hsm.\n")
+    if d.getVar("GARAGE_TARGET_URL") and re.match("^(https?|ftp|file)://.+$", d.getVar("GARAGE_TARGET_URL")) is None:
+        status.addresult("GARAGE_TARGET_URL is set to a bad url.\n")
+    if d.getVar("SOTA_POLLING_SEC") and re.match("^(0|\+?[1-9][0-9]*)$", d.getVar("SOTA_POLLING_SEC")) is None:
```

Comment:
Was this trying to prevent "0"? "0" works fine for me, and I think it's actually valid, albeit questionably desirable. Can we just verify that the value is an integer and nonnegative?